Source/Diagnostics/ReducedDiags/BeamRelevant.cpp

```diff
@@ -0,0 +1,389 @@
+            // write header row
+            #if (AMREX_SPACEDIM == 3)
+            ofs << "#";
+            ofs << "[1]step";             ofs << m_sep;
+            ofs << "[2]time(s)";          ofs << m_sep;
+            ofs << "[3]x_mean(m)";        ofs << m_sep;
+            ofs << "[4]y_mean(m)";        ofs << m_sep;
+            ofs << "[5]z_mean(m)";        ofs << m_sep;
+            ofs << "[6]px_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[7]py_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[8]pz_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[9]gamma_mean";       ofs << m_sep;
+            ofs << "[10]x_rms(m)";        ofs << m_sep;
+            ofs << "[11]y_rms(m)";        ofs << m_sep;
+            ofs << "[12]z_rms(m)";        ofs << m_sep;
+            ofs << "[13]px_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[14]py_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[15]pz_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[16]gamma_rms";       ofs << m_sep;
+            ofs << "[17]emittance_x(m)";  ofs << m_sep;
+            ofs << "[18]emittance_y(m)";  ofs << m_sep;
+            ofs << "[19]emittance_z(m)";  ofs << std::endl;
+            #elif (AMREX_SPACEDIM == 2)
+            ofs << "#";
+            ofs << "[1]step";             ofs << m_sep;
+            ofs << "[2]time(s)";          ofs << m_sep;
+            ofs << "[3]x_mean(m)";        ofs << m_sep;
+            ofs << "[4]z_mean(m)";        ofs << m_sep;
+            ofs << "[5]px_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[6]py_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[7]pz_mean(kg m/s)";  ofs << m_sep;
+            ofs << "[8]gamma_mean";       ofs << m_sep;
+            ofs << "[9]x_rms(m)";         ofs << m_sep;
+            ofs << "[10]z_rms(m)";        ofs << m_sep;
+            ofs << "[11]px_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[12]py_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[13]pz_rms(kg m/s)";  ofs << m_sep;
+            ofs << "[14]gamma_rms";       ofs << m_sep;
+            ofs << "[15]emittance_x(m)";  ofs << m_sep;
+            ofs << "[16]emittance_z(m)";  ofs << std::endl;
+            #endif
+            // close file
+            ofs.close();
+        }
+    }
+
+}
+// end constructor
+
+// function that compute beam relevant quantities
+void BeamRelevant::ComputeDiags (int step)
+{
+
+    // Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    // get MultiParticleContainer class object
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+
+    // get number of species
+    int const nSpecies = mypc.nSpecies();
+
+    // get species names (std::vector<std::string>)
+    auto const species_names = mypc.GetSpeciesNames();
+
+    // inverse of speed of light squared
+    Real constexpr inv_c2 = 1.0 / (PhysConst::c * PhysConst::c);
+
+    // If 2D-XZ, p.pos(1) is z, rather than p.pos(2).
+    #if (AMREX_SPACEDIM == 3)
```

Comment:
Could you remove indentation of **all** the pre-processor directives?

Reply:
Just did that.